### PR TITLE
Initial contribution of the base PHP Dockerfile

### PR DESCRIPTION
**Pull request by ericwill**
This Dockerfile is responsible for the base PHP image
which will be used in the che-devfile-registry.

It is not sufficient to use the plain upstream PHP
image, so we add some customizations necessary for
all the PHP workflows we support.

See eclipse/che#15854

Signed-off-by: Eric Williams <ericwill@redhat.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+FROM php:7.4-apache
+
+RUN apt-get -y update \
+    && apt-get install -y libicu-dev\
+    	tree \
```
Comment (benoitf) [inline Dockerfile:15]: small remark: indent looks strange

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    && sed -i 's/\/var\/www/\/projects/g'  /etc/apache2/apache2.conf \
+    && sed -i 's/None/All/g' /etc/apache2/sites-available/000-default.conf \
+    && echo "ServerName localhost" | tee -a /etc/apache2/apache2.conf
+
+#add composer
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+
+WORKDIR /projects
+
+CMD sleep infinity
```
Comment (benoitf) [inline Dockerfile:41]: I think it's missing the entrypoint and /etc/passwd etc chown to make it compliant for openshift

Comment (ericwill) [reply Dockerfile:41]: I don't think it's needed as this is only a base image, the che-devfile-registry build script does all that: https://github.com/eclipse/che-devfile-registry/blob/master/arbitrary-users-patch/Dockerfile#L6

Comment (benoitf) [reply Dockerfile:41]: ok it's just look strange to republish a new image from this image again while it could be directly compliant.

**Review by benoitf — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (ericwill) [reply Dockerfile:41]: Well this would be the odd one out, so it would be more work to change it in the devfile registry than just leaving it.

**Pull request merged**